engine/src/main/java/org/destinationsol/game/SolObject.java

```diff
@@ -16,41 +16,187 @@
+     *
+     * @param game Game this object belongs to.
+     */
     void onRemove(SolGame game);
 
+    /**
+     * Called whenever this object should receive damage.
+     * <p>
+     * Whenever the object is hit by some projectile or other sort of damage dealing thing, this method is called on the
+     * object to allow it to perform its damage handling. This usually means just subtracting the damage from object's
+     * health and playing a hit sound, but if the object has no health pool or should be otherwise indestructible, or
+     * invincible against some types of damage, this method can be freely left blank.
+     *
+     * @param dmg      Damage the object receives.
+     * @param game     Game this object belongs to.
+     * @param position Position the object was hit at, if hit by point-based damage. Null if not applicable, such as fire.
+     * @param dmgType  Type of the damage object receives.
+     */
     void receiveDmg(float dmg, SolGame game, @Nullable Vector2 position, DmgType dmgType);
 
+    /**
+     * Denotes whether this object is affected by gravity, this is usually constant.
+     *
+     * @return True if object is affected by gravity, false otherwise.
+     */
     boolean receivesGravity();
 
+    /**
+     * Applies external force to object.
+     * <p>
+     * This method usually just scales the {@code force} by objects mass if {@code acc} is set, and then passes the
+     * force to object's internal {@link Body}.
+     *
+     * @param force Force to apply to the object.
+     * @param game  Game this object belongs to.
+     * @param acc   Whether this objects is accelerating by the force, such as in gravity.
+     */
     void receiveForce(Vector2 force, SolGame game, boolean acc);
 
+    /**
+     * Returns object's position.
+     * <p>
+     * This method usually just returns the vector retrieved from object's internal {@link Body}.
+     *
+     * @return Position of the object.
+     */
     Vector2 getPosition();
 
+    /**
+     * Called when an object is too far from player/camera, and is thus to be converted to {@link FarObject}.
+     * <p>
+     * This method is used for optimization of game's objects, by converting to their more resource friendly versions
+     * when they get too far from player/camera. The {@code FarObject} returned by this method should be usable to get
+     * copy of this object back in reverse. If null is returned by this method, this object is designated to be removed
+     * when it gets too far from player/camera, and instead of recreating from {@code FarObject}, to be generated again.
+     *
+     * @return FarObject representation of this object, or null if object is to be removed.
+     */
     FarObject toFarObject();
 
+    /**
+     * Returns list of all {@link Drawable Drawables} this object has assigned.
+     *
+     * @return List of drawables this object has.
+     */
     List<Drawable> getDrawables();
 
+    /**
+     * Returns object's angle.
```

Comment:
`* Returns the object's angle, in degrees.`

Include all important information (like if the angle is in degrees or radians) in the summary sentence.

Side note: This method should, in my opinion, be renamed to something like `getRotation` or `getOrientation`, because the "angle" of an object makes no sense.

Reply:
I think it should be the other way around, actually. The main method description should include _everything_ -- what this method does, what kind of input it takes, and what kind of output it gives and a description of all possible edge cases. The params/returns section must also include information with the same depth of detail, but specific to the parameter/returned value. Necessary or important information may have to be duplicated in the method description and the params/returns description.

Specifically for this case, the `in degrees` detail is very, _very_ important. The summary sentence is the first (and probably only) thing I will read during a quick documentation look-up, and if it does not contain any and all important information about the method, I might create an easily avoidable logic error (by considering that the angle returned is in radians, not degrees, for instance).

Reply:
Usually, when you retrieve this value, you'll only be passing it to some other function expecting angle, or performing some crazy hard logic with it. In the first case, everything is in degrees in DS, in the second place, youo will probably still even look at someimplementations and ignore the javadoc whatsover, to make sure that the method really does what you think it does and what's written in the javadoc

Reply:
Both the reasons you have outlined are the very reasons we need good, well detailed javadoc in the first place -- so that no assumptions have to be made, and time is not wasted by looking up the method source just to find out what the function does. I still do not have a concrete reason to redact my suggestion -- the change recommendation still stands.